File: backend/src/main/java/es/maryshopping/backend/orders/shared/domain/order_line/OrderLine.java
package es.maryshopping.backend.orders.shared.domain.order_line;

import es.maryshopping.backend.orders.shared.domain.order.OrderId;
import lombok.Builder;

import java.math.BigDecimal;

@Builder
public record OrderLine(
        OrderLineId id,
        OrderId orderId,
        CategoryName categoryName,
        ProductId productId,
        ProductName productName,
        ProviderName providerName,
        ProductProviderReference productProviderReference,
        Quantity quantity,
        UnitPrice unitPrice
) {

    public SubTotal subTotal() {
        BigDecimal total = unitPrice.value().multiply(BigDecimal.valueOf(quantity.value()));
        return new SubTotal(total);
    }
}

File: frontend/app/models/orders/OrderLine.ts
// 📦 LÍNEA DE PEDIDO (Estructura de datos)
// Define cómo es cada producto dentro de un pedido:
// - orderLineId, orderId, productId, cantidad, precio unitario
// - subtotal (cantidad × precio)
// - información del producto (nombre, categoría, proveedor)
// Usado para mostrar detalles de cada producto en un pedido

export interface OrderLine {
  orderLineId: string;
  orderId: string;
  productId: string;
  productName: string;
  productProviderReference: string;
  categoryName: string;
  providerName: string;
  quantity: number;
  unitPrice: number;
  subTotal: number;
}
